classes/CartRule.php

```diff
@@ -122,6 +122,8 @@ class CartRuleCore extends ObjectModel
 	 */
 	public function add($autodate = true, $null_values = false)
 	{
+		if (!$this->reduction_currency)
+			$this->reduction_currency = Configuration::get('PS_CURRENCY_DEFAULT'))->id;
```

Comment:
Hi,

What do you want to say about Currency object declaration ?

$reduction_currency is a INT with the Currency ID. This Pull Request is about something like:

```
$cart_rule = new CartRule();
$cart_rule->add();
```

In a module or in an another way, if you don't declare the ID, it will be the default one.

You see ? Or I don't understand what you want to say to me :)